core/01_core.js

```diff
@@ -233,6 +233,43 @@
   }
   const InterruptedPrototype = Interrupted.prototype;
 
+  const promiseHooks = {
+    init: [],
+    before: [],
+    after: [],
+    resolve: [],
+    hasBeenSet: false,
+  };
+
+  function setPromiseHooks(init, before, after, resolve) {
+    if (init) promiseHooks.init.push(init);
```

Comment:
I think the changes in this file should use primordials (ie `ArrayPrototypePush()` instead of calling `.push()`).